core/src/main/java/org/infinispan/util/concurrent/LockTimeoutException.java

```diff
@@ -0,0 +1,83 @@
+package org.infinispan.util.concurrent;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.io.Serializable;
+import java.util.Optional;
+
+import org.infinispan.commands.CommandInvocationId;
+
+/**
+ * Thrown when lock acquisition fails due to timeout.
+ *
+ * @author Pedro Ruivo
+ * @since 12.1
+ */
+public class LockTimeoutException extends TimeoutException implements Serializable {
+
+   private static final byte GENERIC = 0;
+   private static final byte NULL = 1;
+   private static final byte COMMAND_ID = 3;
+
+   private static final long serialVersionUID = 4312558309429074260L;
+   private final Object lockOwner;
+
+   public LockTimeoutException(String msg, Object lockOwner) {
+      //I wish we could drop the stack trace but jboss-logging access it via getStackTrace() and fails.
```

Comment:
> OutdatedTopologyException uses super(message, null, false, false), so I think it should work

OutdatedTopologyException isn't created by jboss-logging. jboss-logging removes itself from the stack trace and if we don't have it, it throws an exception:

```
        final StackTraceElement[] st = result.getStackTrace();
        result.setStackTrace(Arrays.copyOfRange(st, 1, st.length));
        return result;
```

> OTOH if there's no point in logging the stack trace of a lock timeout exception, then what point is there in logging the stack trace of any of our exceptions?

so, you are saying there is useful information in the LockTimeoutException stack trace? the exception is only instantiated in a single place: by a thread in `timeout-executor`

Reply:
> so, you are saying there is useful information in the LockTimeoutException stack trace? the exception is only instantiated in a single place: by a thread in `timeout-executor`

I meant it more like we don't need a comment about the stack trace, because the situation of LockTimeoutException isn't that different from the situation of TimeoutException and perhaps others.

> OutdatedTopologyException isn't created by jboss-logging.

Yeah, I should have checked :)